src/ParameterRegistry.cpp

```diff
@@ -0,0 +1,600 @@
+
+#include "ParameterRegistry.h"
+#include "Logger.h"
+#include "userConfig.h"
+#include <algorithm>
+
+ParameterRegistry ParameterRegistry::_singleton;
+
+// Global variables, needed for backwards compatibility
```

Comment:
Backwards to what? Older versions? How would they interfere here?
Backwards to still-existing legacy code in `main`?

Reply:
The latter, yes. We should probably move away from global variables in the long run.